units/recalculate_change_points.go

```diff
@@ -113,24 +113,58 @@ func (j *recalculateChangePointsJob) Run(ctx context.Context) {
 		}
 
 		result, err := j.changePointDetector.DetectChanges(ctx, floatSeries)
+		if err != nil {
+			j.AddError(errors.Wrapf(err, "Unable to detect change points in time perfData %s", j.PerformanceResultId))
+			return
+		}
 
 		var changePoints []model.ChangePoint
-		for _, pointIndex := range result {
-			mapped := model.CreateChangePoint(pointIndex+latestTriagedChangePointIndex, perfData.Measurement, j.changePointDetector.Algorithm().Name(), j.changePointDetector.Algorithm().Version(), algorithmConfigurationToOptions(j.changePointDetector.Algorithm().Configuration()))
+		for i, pointIndex := range result {
+			var lowerBound int
+			var upperBound int
+
+			if i == 0 {
+				lowerBound = 0
+			} else {
+				lowerBound = result[i-1]
+			}
+
+			if i == len(result)-1 {
+				upperBound = len(floatSeries)
+			} else {
+				upperBound = result[i+1]
+			}
+
+			lowerWindow := floatSeries[lowerBound:pointIndex]
+			upperWindow := floatSeries[pointIndex:upperBound]
+			percentChange := calculatePercentChange(lowerWindow, upperWindow)
+
+			mapped := model.CreateChangePoint(pointIndex+latestTriagedChangePointIndex, perfData.Measurement, j.changePointDetector.Algorithm().Name(), j.changePointDetector.Algorithm().Version(), algorithmConfigurationToOptions(j.changePointDetector.Algorithm().Configuration()), percentChange)
 			changePoints = append(changePoints, mapped)
 		}
 
-		if err != nil {
-			j.AddError(errors.Wrapf(err, "Unable to detect change points in time perfData %s", j.PerformanceResultId))
-			return
-		}
 		mappedChangePoints[perfData.Measurement] = changePoints
 	}
 
 	j.AddError(model.ReplaceChangePoints(ctx, j.env, performanceData, mappedChangePoints))
 	j.AddError(model.MarkPerformanceResultsAsAnalyzed(ctx, j.env, performanceData.PerformanceResultId))
 }
 
+func calculatePercentChange(lowerWindow, upperWindow []float64) float64 {
+	avgLowerWindow := getAverageValue(lowerWindow)
+	avgUpperWindow := getAverageValue(upperWindow)
+
+	return 100 * ((avgUpperWindow / avgLowerWindow) - 1)
+}
+
+func getAverageValue(values []float64) float64 {
```

Comment:
it seems like these math functions should use library functions, I think we already vendor the following which has a function for this: https://godoc.org/github.com/aclements/go-moremath/stats

We also have/can have gonum and other math libraries and there's no real reason to avoid using one (in most cases.)